kernel/mempool.c

```diff
@@ -301,6 +301,11 @@ int k_mem_pool_alloc(struct k_mem_pool *p, struct k_mem_block *block,
 	while (1) {
 		ret = pool_alloc(p, block, size);
 
+		/* [Sanechips] When race condition causes failure, re_alloc. */
```

Comment:
why is the text "[sanechips]" here.
Please also describe in the commit message what the race condition is, under what circumstances does it occur, why this fixes it, it's not completely obvious.